components/stack/stack.vue

```diff
@@ -0,0 +1,286 @@
+          margin-top: 0;
+          margin-inline: 0 var(--stack-gap);
+          margin-bottom: 0;
+        }
+      }
+    }
+  }
+
+  &--lg{
+    @media screen and (max-width: 980px) {
+      &--column {
+        flex-direction: column;
+        & > :not(style) ~ :not(style) {
+          margin-top: var(--stack-gap);
+          margin-inline: 0;
+          margin-bottom: 0;
+        }
+      }
+
+      &--column-reverse {
+        flex-direction: column-reverse;
+        & > :not(style) ~ :not(style) {
+          margin-top: 0;
+          margin-inline: 0;
+          margin-bottom: var(--stack-gap);
+        }
+      }
+
+      &--row {
+        flex-direction: row;
+        & > :not(style) ~ :not(style) {
+          margin-top: 0;
+          margin-inline: var(--stack-gap) 0;
+          margin-bottom: 0;
+        }
+      }
+
+      &--row-reverse {
+        flex-direction: row-reverse;
+        & > :not(style) ~ :not(style) {
+          margin-top: 0;
+          margin-inline: 0 var(--stack-gap);
+          margin-bottom: 0;
+        }
+      }
+    }
+  }
+
+  &--md{
+    @media screen and (max-width: 640px) {
+      &--column {
+        flex-direction: column;
+        & > :not(style) ~ :not(style) {
+          margin-top: var(--stack-gap);
+          margin-inline: 0;
+          margin-bottom: 0;
+        }
+      }
+
+      &--column-reverse {
+        flex-direction: column-reverse;
+        & > :not(style) ~ :not(style) {
+          margin-top: 0;
+          margin-inline: 0;
+          margin-bottom: var(--stack-gap);
+        }
+      }
+
+      &--row {
+        flex-direction: row;
+        & > :not(style) ~ :not(style) {
+          margin-top: 0;
+          margin-inline: var(--stack-gap) 0;
+          margin-bottom: 0;
+        }
+      }
+
+      &--row-reverse {
+        flex-direction: row-reverse;
+        & > :not(style) ~ :not(style) {
+          margin-top: 0;
+          margin-inline: 0 var(--stack-gap);
+          margin-bottom: 0;
+        }
+      }
+    }
+  }
+
+  &--sm{
+    @media screen and (max-width: 480px) {
+      &--column {
+        flex-direction: column;
+        & > :not(style) ~ :not(style) {
+          margin-top: var(--stack-gap);
+          margin-inline: 0;
+          margin-bottom: 0;
+        }
+      }
+
+      &--column-reverse {
+        flex-direction: column-reverse;
+        & > :not(style) ~ :not(style) {
+          margin-top: 0;
+          margin-inline: 0;
+          margin-bottom: var(--stack-gap);
+        }
+      }
+
+      &--row {
+        flex-direction: row;
+        & > :not(style) ~ :not(style) {
+          margin-top: 0;
+          margin-inline: var(--stack-gap) 0;
+          margin-bottom: 0;
+        }
+      }
+
+      &--row-reverse {
+        flex-direction: row-reverse;
+        & > :not(style) ~ :not(style) {
+          margin-top: 0;
+          margin-inline: 0 var(--stack-gap);
+          margin-bottom: 0;
+        }
+      }
+    }
+  }
+}
```

Comment:
I think the CSS definitely needs some refining here.

We should probably be using flexbox gap instead of margin, it has been supported in all major browsers since 2021 and will prevent you from having to do this crazy margin swapping stuff.

Here's an example from the story using [gap for a stack component](https://www.leereamsnyder.com/blog/row-stack-space-layout-components). There's also an every-layout article but I think it's a bit dated and was from before gap was supported in browsers.

Reply:
@braddialpad,

My instinct was the style would ultimately be in `dialtone.css` (`.d-stack`) and then used by `dt-stack`. Whatchathink?

Looking forward, I can see this refactoring existing (and future) components. And _definitely_ throughout Vue components and recipes. Left sidebar is actually a great early destination for it. 

Like I quoted before: _Once you think in stacks, you'll see them everywhere_. Here's a sidebar row you're currently working...

```
<dt-stack gap="300">
  <dt-avatar />
  <dt-stack>
    <div>Ben Parker</div>
    <div>In a meeting</div>
  </dt-stack>
  <dt-badge />
</dt-stack>
```

Reply:
> My instinct was the style would ultimately be in `dialtone.css` (`.d-stack`) and then used by `dt-stack`. Whatchathink?

Yep I think this was actually Ignacio's intention also, he just started out with the Vue component and added the styles there meaning to move them to dialtone later

Reply:
Coolness. In that case, name is `.d-stack`.

Reply:
Yep, as Brad said it was here just to show the proper behaviour, will move the style to dialtone as `d-stack`